diff --git a/node_modules/@graphql-mesh/graphql/index.js b/node_modules/@graphql-mesh/graphql/index.js
index efc197b..21e3d00 100644
--- a/node_modules/@graphql-mesh/graphql/index.js
+++ b/node_modules/@graphql-mesh/graphql/index.js
@@ -498,7 +498,12 @@ class GraphQLHandler {
             schema: nonExecutableSchema,
             executor: async (params) => {
                 const executor = await getExecutorForParams(params, operationHeadersFactory, endpointFactory);
-                return executor(params);
+                params.document.definitions[0].selectionSet.selections = params.document.definitions[0].selectionSet.selections.filter(selection => selection?.name?.value !== '__typename')
+                const result = await executor(params);
+                if (result.data) {
+                    result.data.__typename = 'Query';
+                }
+                return result;
             },
             batch: 'batch' in this.config ? this.config.batch : true,
         };
